### PR TITLE
Fix/complex objects relations

### PR DESCRIPTION
**Pull request by carpawell**

### CODE REVIEW

File: pkg/services/object/tombstone/verify.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+		if err != nil {
+			return fmt.Errorf("heading %s object that was searched: %w", addr, err)
+		}
+
+		if len(h.Children()) != 0 {
+			return nil
+		}
+	}
+
+	return errors.New("no link object found")
```
Comment (roman-khimov) [inline pkg/services/object/tombstone/verify.go:140]: No link here -> can delete.

File: pkg/services/object/tombstone/verify.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+		return nil
+	}
+
+	// it is a child object removal somehow; only allowed for incomplete object put
+	// (garbage collecting) so check if there is a correct link object for the chain
+
+	if sID != nil {
+		err = v.verifyV1Child(ctx, cnr, *sID)
+		if err != nil {
+			return fmt.Errorf("verify V1 split: %w", err)
```
Comment (roman-khimov) [inline pkg/services/object/tombstone/verify.go:96]: But an error is returned here.

**Review by roman-khimov — COMMENTED** [review]
Some tests would be helpful.

File: pkg/services/object/tombstone/verify.go (hunk 1 above)
Comment (carpawell) [reply pkg/services/object/tombstone/verify.go]: fixed

**Comment by carpawell** [comment]
> Some tests would be helpful.

Added some.

**Review by roman-khimov — APPROVED** [review]
Let's try it.

**Pull request merged**